script/version-utils.js

```diff
@@ -0,0 +1,126 @@
+      return verArr.join('.');
+    case 3: // Major
+      if (isBeta) {
+        return currentVersion.split('-')[0];
+      }
+      verArr[0] = Number(verArr[0]) + 1;
+      return verArr.join('.');
+    case 4: // Beta (e.g., 3.0.0-beta.0)
+      if (isBeta) { // Already a beta version then just increase the build number at the end
+        verArr[3] = Number(verArr[3]) + 1;
+        return verArr.join('.');
+      }
+      verArr[0] = Number(verArr[0]) + 1;
+      return verArr.join('.') + '-beta.0';
+    default:
+      logger.error(`ERROR: Invalid input: ${versionIncrement}`);
+      return undefined;
+  }
+};
+
+const updateChangelog = (newVersion) => {
+  logger.log(`Updating CHANGELOG.md with a new release entry - ${newVersion}`);
+  const filePath = path.resolve(__dirname, '../CHANGELOG.md');
+  let changeLog = fs.readFileSync(filePath).toString();
+  const latestEntryIndex = changeLog.indexOf('## [');
+  const newEntry = `## [${newVersion}] - ${new Date().toISOString().slice(0, 10)}
+    \n### Added
+    \n### Removed
+    \n### Changed
+    \n### Fixed
+    \n`;
+  changeLog = changeLog.substring(0, latestEntryIndex) + newEntry + changeLog.substring(latestEntryIndex);
+  fs.writeFileSync(filePath, changeLog);
+};
+
+const updateBaseBranch = (branchName) => {
+  logger.log(`Updating the base branch in .base-branch to ${branchName}`);
+  const filePath = path.resolve(__dirname, '../.base-branch');
+  fs.writeFileSync(filePath, `origin/${branchName}`);
+}
+
+const versionBump = async (option, branchName) => {
+  if (!option) {
+    logger.log('Choose one of the following options to bump the next version:');
+    logger.log('  1. Patch');
+    logger.log('  2. Minor');
+    logger.log('  3. Major');
+    logger.log('  4. Beta');
+    option = Number(await prompt(''));
+  }
+  const newVersion = getNewVersion(currentVersion, option);
+  if (!newVersion) {
+    quit(1);
+  }
+
+  branchName = branchName ? branchName : 'version-bump';
+
+  const prevReleaseBranch = !isPreRelease(currentVersion) && (option === 3 || option === 4)
+    ? `release-${currentVersion.split('.')[0]}.x`
+    : '';
+
+  logger.warn('Warning: you are bumping the version\n');
+  logger.warn(`  From: ${currentVersion}\n`);
+  logger.warn(`  To: ${newVersion}\n`);
+  if (prevReleaseBranch) {
+    logger.warn(`  This will also create ${prevReleaseBranch} branch.`);
+  }
+  await shouldContinuePrompt();
+
+  if (prevReleaseBranch) {
+    const currentBranch = (spawnOrFail('git', [' branch --show-current'], { skipOutput: true })).trim();
+    spawnOrFail('git', [`checkout -b ${prevReleaseBranch}`]);
+    updateBaseBranch(prevReleaseBranch);
+    spawnOrFail('git', ['add -A']);
+    spawnOrFail('git', [`commit -m "Update base branch to ${prevReleaseBranch}"`]);
+    spawnOrFail('git', [`push origin HEAD:${prevReleaseBranch} -f`]);
+    logger.log(`Branch ${prevReleaseBranch} is created. Please make sure to set branch protection.`);
+
+    //Switch back to the local branch
```

Comment:
nit: Few more comments left to update "// " :)

Reply:
Thanks. Hopefully addressed all of these now :D